docs/device_api.md

```diff
@@ -346,36 +357,71 @@ PUT /v2/entities/[ENTITY_ID]/attrs/ping
 
 ```
 
-ContextBroker API is quite flexible and allows to update an attribute in several ways. Please have a look to
-the [NGSIv2 specification](http://telefonicaid.github.io/fiware-orion/api/v2/stable) for details.
+ContextBroker API is quite flexible and allows to update an attribute in several ways. Please have a look to the [NGSIv2 specification](http://telefonicaid.github.io/fiware-orion/api/v2/stable) for details.
 
-**Important note**: don't use operations in the NGSI API with creation semantics. Otherwise, the entity/attribute
-will be created locally to ContextBroker and the command will not progress to the device (and you will need to delete
-the created entity/attribute if you want to make it to work again). Thus, the following operations *must not* be used:
+**Important note**: don't use operations in the NGSI API with creation semantics. Otherwise, the entity/attribute will be created locally to ContextBroker and the command will not progress to the device (and you will need to delete the created entity/attribute if you want to make it to work again). Thus, the following operations *must not* be used:
 
 * `POST /v2/entities`
 * `PUT /v2/entities`
 * `POST /v2/op/entites` with `actionType` `append`, `appendStrict` or `replace`
 * `POST /v1/updateContext` with `actionType` `APPEND`, `APPEND_STRICT` or `REPLACE`
 
-For HTTP devices, whose "endpoint": "http://[DEVICE_IP]:[PORT]" is declared, then your device is supposed to be
-listening for commands at that URL in a synchronous way.
+## Push commands
+
+Push commands are those that are sent to the device once the IoT Agent receives the request from the Context Broker. In order to send push commands it is needed to set the "endpoint": "http://[DEVICE_IP]:[PORT]" in the device provision. The device is supposed to be listening for commands at that URL in a synchronous way. Make sure the device endpoint is reachable by the IoT Agent.
+
+Once the command is delivered, the device should return the result of the command to the IoT Agent as the answer to the 
+HTTP request. This result will be progressed to the Context Broker where it will be stored in the "command_info" attribute. The status of the command will be stored in the "command_status" attribute (`OK` if everithing goes right). 
+
+Push commands are only valid for HTTP devices. For MQTT devices it is not needed to set the "endpoint" paramater. 
```

Comment:
In the case of MQTT we don't have a endpoint but IoTAgent is able to progress the command directly to the device without storing. I understand that this is also a push command.

I mean, under my understanding:

* "it involes storing command to be recovered later by the device" -> "pull command".
* "it is delivered inmediatelly without storing at IoTA" -> "push command"

Is my interpretation correct?

Reply:
After discussing, what could make sense is to distinguish first by protocol (i.e. "HTTP commands" and  "MQTT commands") and then, within HTTP commands, distinguish between "Pull" and "Push". The pull vs. push discussion wouldn't have sense in the context of MQTT protocol.

Reply:
Maybe the correct approach is to consider pull and push commands only in HTTP. MQTT would have just commands. To change approach in the next pull request